lightning/src/ln/interactivetxs.rs

```diff
@@ -822,11 +865,170 @@ impl StateMachine {
 	]);
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum AddingRole {
+	Local,
+	Remote,
+}
+
+/// Represents an input -- local or remote (both have the same fields)
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct LocalOrRemoteInput {
+	serial_id: SerialId,
+	input: TxIn,
+	prev_output: TxOut,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum InteractiveTxInput {
+	Local(LocalOrRemoteInput),
+	Remote(LocalOrRemoteInput),
+	// TODO(splicing) SharedInput should be added
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SharedOwnedOutput {
+	tx_out: TxOut,
+	local_owned: u64,
+}
+
+impl SharedOwnedOutput {
+	fn new(tx_out: TxOut, local_owned: u64) -> SharedOwnedOutput {
+		debug_assert!(local_owned <= tx_out.value, "SharedOwnedOutput: Inconsistent local_owned value {}, larger than output value {}", local_owned, tx_out.value);
+		SharedOwnedOutput { tx_out, local_owned }
+	}
+
+	fn remote_owned(&self) -> u64 { 
+		self.tx_out.value.saturating_sub(self.local_owned)
+	}
+}
+
+/// Represents an output, with information about
+/// its control -- exclusive by the adder or shared --, and
+/// its ownership -- fully owned by the adder or jointly
```

Comment:
I think we need to expand on the meaning of "owned" here. I assume it relates to the value. So "value fully owned" might be clearer at least in the docs. The variant naming can remain.

Reply:
Yes, ownership here refers to whom the satoshis belong. Done.